.github/workflows/security.yaml

```diff
@@ -32,7 +35,60 @@ jobs:
           api-key: ${{ secrets.FOSSA_API_KEY }}
 
       - name: Run FOSSA Test
-        uses: fossas/fossa-action@main
-        with:
-          api-key: ${{ secrets.FOSSA_API_KEY }}
-          run-tests: true
+        id: fossa_test
+        env:
+          FOSSA_API_KEY: ${{ secrets.FOSSA_API_KEY }}
+        run: |
+          MAIN_BRANCH="refs/heads/main"
+          CURRENT_BRANCH="${{ github.ref }}"
+          DIFF_OPTION=""
+      
+          if [ "$CURRENT_BRANCH" != "$MAIN_BRANCH" ]; then
+            git fetch origin main
+            MAIN_SHA=$(git rev-parse origin/main)
```

Comment:
If we're using $MAIN_BRANCH elsewhere, maybe we should use it here too?

```bash
git fetch origin $MAIN_BRANCH
MAIN_SHA=$(git rev-parse origin/$MAIN_BRANCH)
```

this is absolutely a nitpick

Reply:
I can't do that because the `git rev-parse` fails:

```sh
git rev-parse origin/refs/heads/main
origin/refs/heads/main
fatal: ambiguous argument 'origin/refs/heads/main': unknown revision or path not in the working tree.
```

Hence I'm refactoring to $MAIN_BRANCH and $MAIN_BRANCH_REF like this:
```
          MAIN_BRANCH="main"
          MAIN_BRANCH_REF="refs/heads/$MAIN_BRANCH"
          CURRENT_BRANCH_REF="${{ github.ref }}"
          DIFF_OPTION=""

          if [ "$CURRENT_BRANCH_REF" != "$MAIN_BRANCH_REF" ]; then
            git fetch origin $MAIN_BRANCH
            MAIN_SHA=$(git rev-parse origin/$MAIN_BRANCH)
            DIFF_OPTION="--diff $MAIN_SHA"
          fi
```